keyboards/keycapsss/kimiko/config.h

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2019 MechMerlin
+ * Copyright 2023 @Ex3c4Def
+ * Copyright 2023 @BenRoe (keycapsss.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// OLED definitions
+#ifndef OLED_FONT_H
+#   define OLED_FONT_H "keyboards/keycapsss/kimiko/glcdfont.c"
+#endif
```

Comment:
This is causing issues with the existing keymaps.  You should be good to rename the file to `post_config.h`, and that should fix the issue.

Reply:
Renamed `keyboards/keycapsss/kimiko/config.h` to `post_config.c`.
Get an error if i try to compile the Rev1 default keymap.

<details>
  <summary>Error details</summary>

  ```bash
  ❯ qmk compile -kb keycapsss/kimiko/rev1 -km default 
  Ψ Compiling keymap with gmake --jobs=1 keycapsss/kimiko/rev1:default
  
  
  QMK Firmware 0.21.6
  Making keycapsss/kimiko/rev1 with keymap default
  
  Generating: .build/obj_keycapsss_kimiko_rev1_default/src/info_deps.d                                [OK]
  avr-gcc (Homebrew AVR GCC 8.5.0_1) 8.5.0
  Copyright (C) 2018 Free Software Foundation, Inc.
  This is free software; see the source for copying conditions.  There is NO
  warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  
  Generating: .build/obj_keycapsss_kimiko_rev1/src/info_config.h                                      [OK]
  Generating: .build/obj_keycapsss_kimiko_rev1/src/default_keyboard.c                                 [OK]
  Generating: .build/obj_keycapsss_kimiko_rev1/src/default_keyboard.h                                 [OK]
  Compiling: keyboards/keycapsss/kimiko/kimiko.c                                                      [OK]
  Compiling: .build/obj_keycapsss_kimiko_rev1/src/default_keyboard.c                                  [OK]
  Compiling: quantum/keymap_introspection.c                                                           [OK]
  Compiling: quantum/quantum.c                                                                        [OK]
  Compiling: quantum/bitwise.c                                                                        [OK]
  Compiling: quantum/led.c                                                                            [OK]
  Compiling: quantum/action.c                                                                         [OK]
  Compiling: quantum/action_layer.c                                                                   [OK]
  Compiling: quantum/action_tapping.c                                                                 [OK]
  Compiling: quantum/action_util.c                                                                    [OK]
  Compiling: quantum/eeconfig.c                                                                       [OK]
  Compiling: quantum/keyboard.c                                                                       [OK]
  Compiling: quantum/keymap_common.c                                                                  [OK]
  Compiling: quantum/keycode_config.c                                                                 [OK]
  Compiling: quantum/sync_timer.c                                                                     [OK]
  Compiling: quantum/logging/debug.c                                                                  [OK]
  Compiling: quantum/logging/sendchar.c                                                               [OK]
  Compiling: quantum/bootmagic/magic.c                                                                [OK]
  Compiling: quantum/matrix_common.c                                                                  [OK]
  Compiling: quantum/matrix.c                                                                         [OK]
  Compiling: quantum/debounce/sym_defer_g.c                                                           [OK]
  Compiling: quantum/split_common/split_util.c                                                        [OK]
  Compiling: quantum/split_common/transport.c                                                         [OK]
  Compiling: quantum/split_common/transactions.c                                                      [OK]
  Compiling: quantum/main.c                                                                           [OK]
  Compiling: quantum/mousekey.c                                                                       [OK]
  Compiling: quantum/color.c                                                                          [OK]
  Compiling: quantum/rgblight/rgblight.c                                                              [OK]
  Compiling: quantum/process_keycode/process_rgb.c                                                    [OK]
  Compiling: platforms/avr/drivers/ws2812_bitbang.c                                                   [OK]
  Compiling: quantum/led_tables.c                                                                     [OK]
  Compiling: quantum/crc.c                                                                            [OK]
  Compiling: drivers/oled/oled_driver.c                                                               [OK]
  Compiling: quantum/process_keycode/process_magic.c                                                  [OK]
  Compiling: quantum/send_string/send_string.c                                                        [OK]
  Compiling: quantum/encoder.c                                                                        [OK]
  Compiling: quantum/process_keycode/process_grave_esc.c                                              [OK]
  Compiling: quantum/process_keycode/process_space_cadet.c                                            [OK]
  Compiling: platforms/avr/drivers/i2c_master.c                                                       [OK]
  Archiving: .build/obj_keycapsss_kimiko_rev1_default/i2c_master.o                                    [OK]
  Compiling: platforms/avr/drivers/i2c_slave.c                                                        [OK]
  Archiving: .build/obj_keycapsss_kimiko_rev1_default/i2c_slave.o                                     [OK]
  Compiling: platforms/avr/drivers/serial.c                                                           [OK]
  Archiving: .build/obj_keycapsss_kimiko_rev1_default/serial.o                                        [OK]
  Assembling: platforms/avr/xprintf.S                                                                 [OK]
  Compiling: platforms/avr/printf.c                                                                   [OK]
  Compiling: tmk_core/protocol/host.c                                                                 [OK]
  Compiling: tmk_core/protocol/report.c                                                               [OK]
  Compiling: tmk_core/protocol/usb_device_state.c                                                     [OK]
  Compiling: tmk_core/protocol/usb_util.c                                                             [OK]
  Compiling: platforms/suspend.c                                                                      [OK]
  Compiling: platforms/synchronization_util.c                                                         [OK]
  Compiling: platforms/timer.c                                                                        [OK]
  Compiling: platforms/avr/hardware_id.c                                                              [OK]
  Compiling: platforms/avr/platform.c                                                                 [OK]
  Compiling: platforms/avr/suspend.c                                                                  [OK]
  Compiling: platforms/avr/timer.c                                                                    [OK]
  Compiling: platforms/avr/bootloaders/dfu.c                                                          [OK]
  Compiling: tmk_core/protocol/lufa/lufa.c                                                            [OK]
  Compiling: tmk_core/protocol/usb_descriptor.c                                                       [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Class/Common/HIDParser.c                                       [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Device_AVR8.c                                        [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/EndpointStream_AVR8.c                                [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Endpoint_AVR8.c                                      [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Host_AVR8.c                                          [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/PipeStream_AVR8.c                                    [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/Pipe_AVR8.c                                          [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/USBController_AVR8.c                                 [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/AVR8/USBInterrupt_AVR8.c                                  [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/ConfigDescriptors.c                                       [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/DeviceStandardReq.c                                       [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/Events.c                                                  [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/HostStandardReq.c                                         [OK]
  Compiling: lib/lufa/LUFA/Drivers/USB/Core/USBTask.c                                                 [OK]
  Compiling: tmk_core/protocol/lufa/usb_util.c                                                        [OK]
  Linking: .build/keycapsss_kimiko_rev1_default.elf                                                   [ERRORS]
   | 
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `oled_init_user'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_space'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_mod_status_gui_alt'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_mod_status_ctrl_shift'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_logo'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_layer_state'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_status_main'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | /opt/homebrew/opt/avr-binutils/bin/avr-ld: .build/obj_keycapsss_kimiko_rev1_default/quantum/keymap_introspection.o (symbol from plugin): in function `layer_state_set_user':
   | (.text+0x0): multiple definition of `render_status_secondary'; .build/obj_keycapsss_kimiko_rev1_default/keyboards/keycapsss/kimiko/kimiko.o (symbol from plugin):(.text+0x0): first defined here
   | collect2: error: ld returned 1 exit status
   | 
  gmake[1]: *** [builddefs/common_rules.mk:268: .build/keycapsss_kimiko_rev1_default.elf] Error 1
  gmake: *** [Makefile:392: keycapsss/kimiko/rev1:default] Error 1
  Make finished with errors
  ```
</details>

Reply:
@drashna my fault. Pulled the latest changes from `master` and `qmk compile -kb keycapsss/kimiko/rev2 -km default` compiles without an error.
But get an error if i compile Rev1 `qmk compile -kb keycapsss/kimiko/rev1 -km default`

<details>
  <summary>Error Message</summary>

  ```bash
❯ qmk compile -kb keycapsss/kimiko/rev1 -km default                                                   
Ψ Compiling keymap with gmake --jobs=1 keycapsss/kimiko/rev1:default


QMK Firmware 0.22.3
Making keycapsss/kimiko/rev1 with keymap default

avr-gcc (Homebrew AVR GCC 8.5.0_1) 8.5.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: .build/obj_keycapsss_kimiko_rev1_default/src/default_keyboard.c                         In file included from .build/obj_keycapsss_kimiko_rev1_default/src/default_keyboard.c:26:
./.build/obj_keycapsss_kimiko_rev1_default/src/default_keyboard.h:32:2: error: #error ("<keyboard>.h is required unless your keyboard uses data-driven configuration. Please rename your keyboard's header file to <keyboard>.h")
 #error("<keyboard>.h is required unless your keyboard uses data-driven configuration. Please rename your keyboard's header file to <keyboard>.h")
  ^~~~~
 [ERRORS]
 | 
 | 
 | 
gmake[1]: *** [builddefs/common_rules.mk:361: .build/obj_keycapsss_kimiko_rev1_default/.build/obj_keycapsss_kimiko_rev1_default/src/default_keyboard.o] Error 1
Make finished with errors
gmake: *** [Makefile:392: keycapsss/kimiko/rev1:default] Error 1
  ```
</details>